types/coin.go

```diff
@@ -690,9 +690,12 @@ func (coins Coins) Empty() bool {
 	return len(coins) == 0
 }
 
-// AmountOf returns the amount of a denom from coins
+// AmountOf returns the amount of a denom from coins.  Will return 0 if the denom is invalid.
 func (coins Coins) AmountOf(denom string) math.Int {
-	mustValidateDenom(denom)
+	if err := ValidateDenom(denom); err != nil {
+		return math.ZeroInt()
+	}
```

Comment:
was this meant to be added back? DecCoins also had its `mustValidateDenom` call removed. i jumped around the codepaths for these methods, and i wasn't able to find any paths where the denom wasn't already validated.